common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationRequest.java

```diff
@@ -22,20 +22,62 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.providers.oauth2;
 
+import com.microsoft.identity.common.exception.ClientException;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashSet;
+import java.util.Set;
+
 /**
  * A class holding the state of the Authorization Request (OAuth 2.0).
  * https://tools.ietf.org/html/rfc6749#section-4.1.1
  * This should include all fo the required parameters of the authorization request for oAuth2
  * This should provide an extension point for additional parameters to be set
  */
-public class AuthorizationRequest {
-
+public abstract class AuthorizationRequest {
+    /**
+     * A required value.
+     *
+     * Must include code for the authorization code flow.
+     */
     private String mResponseType;
+
+    /**
+     * A required value.
+     *
+     * The Application ID that the registration portal (apps.dev.microsoft.com) assigned your app.
+     */
     private String mClientId;
+
+    /**
+     * A recommended value.
+     *
+     * The redirect_uri of your app, where authentication responses can be sent and received by your app.
+     * It must exactly match one of the redirect_uris you registered in the portal, except it must be url encoded.
+     */
     private String mRedirectUri;
-    private String mScope;
+
+    /**
+     * A recommended value.
+     *
+     * A value included in the request that will also be returned in the token response.
+     * It can be a string of any content that you wish. A randomly generated unique value is
+     * typically used for preventing cross-site request forgery attacks. The value can also
+     * encode information about the user's state in the app before the authentication request
+     * occurred, such as the page or view they were on.
+     */
     private String mState;
 
+    /**
+     * Scopes scopes that you want the user to consent to is required for V2 auth request.
+     *
+     * The scope is an ignored value for V1 auth request.
+     * For v1 Azure AD apps, scopes must be statically configured in the Azure Portal.
```

Comment:
The decision was taken not to use v1/v2 terminology when referring to app registrations.  There is no v1 or v2 app from a directory point of view.  They are intended to look the same.  You can refer to the v1 and v2 endpoints.  Since this is the subclass of AuthorizationRequest for AAD (which is V1... some of this comment probably isn't needed)